scripts/stac/imagery/generate_metadata.py

```diff
@@ -0,0 +1,176 @@
+    [Digital Surface Model / Digital Elevation Model] within the [region] [?- location] region in [year(s)].
+    Satellite Imagery:
+    Satellite imagery within the [Region] region captured in [Year(s)].
+    Historical Imagery:
+    Scanned aerial imagery within the [Region] region captured in [Year(s)].
+
+    Args:
+        subtype: Dataset subtype description - used to determine if the dataset is imagery, elevation, ect.
+        region: Region of Dataset
+        start_date: Dataset capture start date
+        end_date: Dataset capture end date
+        location: Optional location of dataset, e.g.- Hutt City
+        event: Optional details of capture event, e.g. - Cyclone Gabreille
+        historic_survey_number: Optional historic imagery survey number, e.g.- SNC88445
+
+    Returns:
+        Dataset Description
+    """
+    _validate_no_empty_strings({"region": region, "subtype": subtype})
+
+    date = _format_date_for_metadata(start_date, end_date)
+    location_txt = _format_location_for_elevation_description(location)  # TODO: rename
+    region = _region_map(region)
+    desc = None
+
+    if historic_survey_number:
+        desc = f"Scanned aerial imagery within the {region} region captured in {date}."
+
+    if subtype == "Satellite Imagery":
+        desc = f"Satellite imagery within the {region} region captured in {date}."
+    if subtype in ["Urban Aerial Photos", "Rural Aerial Photos"]:
+        desc = f"Orthophotography within the {region} region captured in the {date} flying season."
+    if subtype == "DEM":
+        desc = " ".join(f"Digital Elevation Model within the {region} {location_txt or ''} region in {date}.".split())
+    if subtype == "DSM":
+        desc = " ".join(f"Digital Surface Model within the {region} {location_txt or ''} region in {date}.".split())
+
+    if not desc:
+        raise SubtypeParameterError(subtype)
+    if event and event != "":
+        desc = desc.replace(".", f", published as a record of the {event} event.")
+
+    return desc
+
+
+def _format_date_for_metadata(start_date: datetime, end_date: datetime) -> str:
+    if start_date.year == end_date.year:
+        return str(start_date.year)
+    return f"{start_date.year} - {end_date.year}"
+
+
+def _format_name_for_title(region: str, location: Optional[str]) -> str:
+    if location and location != "":
+        return location
+    return region
+
+
+def _is_preview(lifecycle: str) -> Optional[str]:
+    """lifeycle is only added to a dataset title if the status is preview."""
+    if lifecycle == "preview":
+        return "- preview"
+    return None
+
+
+def _format_location_for_elevation_description(location: Optional[str]) -> Optional[str]:
+    if location and location != "":
+        return f"- {location}"
+    return None
+
+
+def _format_event_for_elevation_title(event: Optional[str]) -> Optional[str]:
+    if event and event != "":
+        return f"- {event}"
+    return None
+
+
+def _region_map(region: str) -> str:
```

Comment:
Sorry I not sure how this fits into this function, other than to edit the last line of the function but then its making this line more complicated?

Also maybe the naming of it should be changed to `_convert_region_ascii_to_utf8` ?

Reply:
Now that `--region` is an enum choice, you can get rid of all the logic and replacements by using a `dict` lookup:

```python
HUMAN_READABLE_REGIONS = {
    "hawkes-bay": "Hawke's Bay",
    …
}
```

Then, instead of having a `_region_map` you can just do `HUMAN_READABLE_REGIONS[region]`.